### PR TITLE
Refactor offer/trade related classes in core and desktop

### PR DESCRIPTION
**Pull request by ghubstan**
These refactoring changes are for reducing existing and potential duplication coming with the addition of new trading protocol support in the gRPC API.  Some minor styling and logic simplification changes are also included.

- Convert OfferUtil to injected singleton, and move various offer related utility methods into it.

- Delete both MakerFeeProvider classes, which were wrappers around the same static old OfferUtil method.

- Inject OfferUtil into CreateOfferDataModel, CreateOfferViewModel,  TakeOfferDataModel, TakeOfferViewModel, MutableOfferDataModel,  MutableOfferViewModel, OfferDataModel, EditOfferDataModel,  EditOfferViewModel

- Refactor TakeOfferViewModel

	Use OfferUtil, remove unused fields & methods.
	Made minor logic simplification, style and formatting changes.

- MutableOfferDataModel

	Made minor logic simplification, style and formatting changes.

- MutableOfferView uses new paymentAccount.isHalCashAccount().

- MutableOfferViewModel

	Refactored to use new VolumeUtil, CoinUtil, OfferUtil.
	Removed unused fields & accessors.
	Made minor style change.

- Refactored OfferDataModel to use new OfferUtil

- Refactor CreateOfferService

	Inject and use OfferUtil
	Move some utility methods to OfferUtil
	Remove unused fields

- Offer

	Refactored to use new VolumeUtil for volume calculations.
	Made stateProperty and errorMessageProperty fields private.

- PaymentAccount

	Moved isHalCashAccount type check to this class.
	Moved getTradeCurrency logic to this class.

- Contract, TradeStatistics2, TradeStatistics3

	Refactored to use new VolumeUtil for volume calculations.

- Trade

	Refactored to use new VolumeUtil for volume calculations.
	Made minor logic simplification, style and formatting changes.

- CoinUtil

	Moved some coin utility methods into this class

- CoinUtilTest

	Moved (coin related) tests from CoinCryptoUtilsTest and OfferUtilTest
	into CoinUtilTest, and deleted OfferUtilTest, CoinCryptoUtilsTest.

- Adjust create and edit offer tests to model refactoring

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/offer/MutableOfferDataModel.java [hunk 1]
```diff
@@ -133,6 +133,9 @@
     protected boolean allowAmountUpdate = true;
     private final TradeStatisticsManager tradeStatisticsManager;
 
+    private final Predicate<ObjectProperty<Coin>> isPositiveAmount = (c) -> c.get() != null && !c.get().isZero();
+    private final Predicate<ObjectProperty<Price>> isPositivePrice = (p) -> p.get() != null && !p.get().isZero();
+    private final Predicate<ObjectProperty<Volume>> isPositiveVolume = (v) -> v.get() != null && !v.get().isZero();
```
Comment (sqrrm) [inline desktop/src/main/java/bisq/desktop/main/offer/MutableOfferDataModel.java:138]: Shouldn't these be `isNonZero...`?

**Review by sqrrm — COMMENTED** [review]
One minor comment. I think it's otherwise good.

To make it easier to review, it helps to keep refactorings separated in different commits. In particular pure refactor work and minor improvements are very helpful when kept separate.

File: core/src/main/java/bisq/core/offer/CreateOfferService.java [hunk 2]
```diff
@@ -294,25 +274,6 @@ public double getSellerSecurityDepositAsDouble(double buyerSecurityDeposit) {
...
-        } else {
-            return 0;
-        }
-    }
-
-    public boolean isBuyOffer(OfferPayload.Direction direction) {
-        return OfferUtil.isBuyOffer(direction);
-    }
-
-
```
Comment (ghubstan) [inline core/src/main/java/bisq/core/offer/CreateOfferService.java:315]: 3 methods above were moved to OfferUtil.
(test -- can I add inline comment to my own pr?)

File: desktop/src/main/java/bisq/desktop/main/offer/MutableOfferDataModel.java (hunk 1 above)
Comment (ghubstan) [reply desktop/src/main/java/bisq/desktop/main/offer/MutableOfferDataModel.java:138]: Fixed in 2ae6bfa, sub branch [5-suggested-changes](https://github.com/ghubstan/bisq/tree/5-suggested-changes).  Fixes for this PR will go into a separate PR.

**Review by sqrrm — APPROVED** [review]
utACK

**Pull request merged**